src/Sylius/Bundle/ApiBundle/spec/EventSubscriber/ChangePaymentMethodEventSubscriberSpec.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\EventSubscriber;
+
+use ApiPlatform\Action\PlaceholderAction;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\ApiBundle\Command\Account\ChangePaymentMethod;
+use Sylius\Bundle\ApiBundle\Command\Checkout\ChoosePaymentMethod;
+use Sylius\Bundle\ApiBundle\Command\SendContactRequest;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Payment\Repository\PaymentRequestRepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ControllerArgumentsEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+final class ChangePaymentMethodEventSubscriberSpec extends ObjectBehavior
+{
+    function let(PaymentRequestRepositoryInterface $paymentRequestRepository): void
+    {
+        $this->beConstructedWith($paymentRequestRepository);
+    }
+
+    function it_updates_payments_request_for_change_payment_method_command(
+        PaymentRequestRepositoryInterface $paymentRequestRepository,
+        HttpKernelInterface $kernel,
+        PaymentRequestInterface $paymentRequest,
+        PaymentMethodInterface $paymentMethod,
+    ): void {
+        $request = new Request();
+        $request->setMethod(Request::METHOD_PATCH);
+
+        $changePaymentMethod = new ChangePaymentMethod('paypal');
+        $changePaymentMethod->paymentId = 1;
+        $paymentRequestRepository->findAllByPaymentId(1)->willReturn([$paymentRequest]);
+        $paymentRequest->getState()->willReturn(PaymentRequestInterface::STATE_NEW);
+        $paymentRequest->getMethod()->willReturn($paymentMethod);
+        $paymentMethod->getCode()->willReturn('stripe');
+        $paymentRequest->setState(PaymentRequestInterface::STATE_CANCELLED)->shouldBeCalled();
+
+        $this->cancelPaymentRequestsWithDifferentPaymentMethod(new ControllerArgumentsEvent(
+            $kernel->getWrappedObject(),
+            new PlaceholderAction(),
+            [
+                $changePaymentMethod,
+            ],
+            $request,
+            HttpKernelInterface::MASTER_REQUEST,
```

Comment:
```suggestion
            HttpKernelInterface::MAIN_REQUEST,
```
`::MASTER_REQUEST` is deprecated